backend/app/Http/Controllers/MarketplaceController.php

```diff
@@ -54,6 +55,37 @@ public function updateCard($card_id)
         return $this->getCardDetail($card_id);
     }
 
+    public function buyCard($card_id)
+    {
+        //update card user's id with current id
+        $user = auth()->user();
+        $card = Card::find($card_id);
+        if ($card->isUserOwner($user)) {
+            return response()->build("User own the card");
```

Comment:
Lets not return any responses directly like this, + we do not want to return at this point if the user is the owner (we want to continue).